docs/architecture/devops-for-aspnet-developers/actions-deploy.md

```diff
@@ -0,0 +1,491 @@
+          with:
+            creds: ${{ secrets.AZURE_CREDENTIALS }}
+        
+        - name: Swap staging slot into production
+          id: slot_swap
+          run: |
+            az webapp deployment slot swap -g ${{ env.rg-name }} -n ${{ env.app-name }} -s staging
+            url=$(az webapp show -g ${{ env.rg-name }} -n ${{ env.app-name }} --query "defaultHostName" -o tsv)
+            echo "::set-output name=url::http://$url"
+          
+        - name: az cli logout
+          run: az logout
+    ```
+
+    Notice the following things:
+  
+    1. Once again, you specify a new job `deploy_prod` that `needs` `deploy_staging` to complete before starting.
+    1. You're targeting the `PROD` environment this time. Also, the `url` value is different from before.
+    1. For the steps, you don't need to download the artifact since you're just going to perform a slot swap. You start by executing a `login` to the Azure context.
+    1. The `Swap staging slot into production` step is a multi-line `run` command (note the use of the pipe symbol `|`). You also specify an `id` for this step so that you can refer to it (you refer to it in the `url` property of the `environment`). The first line executes the slot swap using the variables you defined above in the workflow. The second line uses an `az webapp show` command to extract the URL of the target web app. This final line uses `::set-output` in an echo to create an output variable for this task, setting the value to the web app URL.
+
+    > [!NOTE]
+    > The URL _must_ start with `http://` or `https://` or it will not render.
+
+1. Commit the file.
+1. Let the workflow run for a couple minutes until it has deployed to `PRE-PROD`. At this point, the workflow will pause and wait for the required approval since you're targeting the `PROD` environment, which requires an approval as defined earlier:
+
+    - ![Waiting for an approval](./media/actions/deploy/waiting-for-approval.jpg)
+    **Figure 9**: Waiting for an approval.
+
+1. Select **Review deployments**, select the **PROD** checkbox, optionally add a comment, and then select **Approve and deploy** to start the `PROD` job.
+
+    - ![Approve the PROD deployment](./media/actions/deploy/approval.jpg)
+    **Figure 10**: Approve the PROD deployment.
+
+1. The deployment should only take a few seconds. Once it has completed, the URL for the `PROD` environment will update.
+
+    - ![PROD deployment completed](./media/actions/deploy/prod-deploy-complete.jpg)
+    **Figure 11**: PROD deployment completed.
+
+1. Selecting the `PROD` URL will navigate you to the `PROD` site.
+
+    - ![The PROD site](./media/actions/deploy/deployed-to-prod.jpg)
+    **Figure 12**: The PROD site.
+
+## Add a manual queue option
+
+You now have an end-to-end build and deploy workflow, including approvals! One more change you can make is to add a manual trigger to the workflow so that the workflow can be triggered from within the **Actions** tab of the repository.
```

Comment:
```suggestion
You now have an end-to-end build and deploy workflow, including approvals. One more change you can make is to add a manual trigger to the workflow so that the workflow can be triggered from within the **Actions** tab of the repository.
```